sql/handler.cc

```diff
@@ -4517,7 +4517,7 @@ void handler::get_dynamic_partition_info(PARTITION_STATS *stat_info,
   stat_info->update_time=          stats.update_time;
   stat_info->check_time=           stats.check_time;
   stat_info->check_sum=            0;
-  if (table_flags() & (HA_HAS_OLD_CHECKSUM | HA_HAS_OLD_CHECKSUM))
+  if (table_flags() & HA_HAS_OLD_CHECKSUM)
```

Comment:
I confirm it should be HA_HAS_OLD_CHECKSUM | HA_HAS_NEW_CHECKSUM.
Since this doesn't seem to be covered by any test case, it would be nice to implement new one.